packages/docusaurus-theme-classic/src/theme/SiteMetadata/index.tsx

```diff
@@ -58,10 +58,14 @@ function AlternateLangHeaders(): JSX.Element {
 // Default canonical url inferred from current page location pathname
 function useDefaultCanonicalUrl() {
   const {
-    siteConfig: {url: siteUrl},
+    siteConfig: {url: siteUrl, trailingSlash = false},
   } = useDocusaurusContext();
   const {pathname} = useLocation();
-  return siteUrl + useBaseUrl(pathname);
+  return (
+    siteUrl +
+    useBaseUrl(pathname).replace(/\/+$/, '') +
+    (trailingSlash ? '/' : '')
```

Comment:
```suggestion
    useBaseUrl(pathname).replace(/\/+$/, trailingSlash ? '/' : '')
```

Reply:
Thank you @Josh-Cena for your review. I accepted your suggestion + ran again `yarn format` to fix a small linter issue.
Will appreciate your second review to merge this fix so we can see it in the next docusaurus patch version.